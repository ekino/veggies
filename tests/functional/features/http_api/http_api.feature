@http_api
Feature: Using fixtures with http API extension

    @yaml
    Scenario: Setting json body from .yaml fixture file
        Given I mock http call to forward request body for path /users/yaml
        And set request json body from yaml_00
        When I POST https://fake.io/users/yaml
        Then response status code should be 200
        And response status should be ok
        And response body should match snapshot

    @yaml
    Scenario: Setting form body from .yaml fixture file
        Given I mock http call to forward request body for path /users/yaml
        And set request form body from yaml_00
        When I POST https://fake.io/users/yaml
        Then response status code should be 200


    Scenario: Setting form body directly
        Given I mock http call to forward request body for path /users/yaml
        And set request form body
            | id         | 1       |
            | first_name | Raphaël |
            | last_name  | Benitte |
            | gender     | male    |
        When I POST https://fake.io/users/yaml
        Then response status code should be 200

    @yaml
    Scenario: Setting json body from .yml fixture file
        Given I mock http call to forward request body for path /users/yml
        And set request json body from yaml_01
        When I POST https://fake.io/users/yml
        Then response status code should be 200

    @yaml
    Scenario: Setting form body from .yml fixture file
        Given I mock http call to forward request body for path /users/yml
        And set request form body from yaml_01
        When I POST https://fake.io/users/yml
        Then response status code should be 200
        And response should match fixture check

    @text
    Scenario: Setting json body from .txt fixture file
        Given I mock http call to forward request body for path /users/txt
        And set request json body from text_00
        When I POST https://fake.io/users/txt
        Then response status code should be 200

    @text
    Scenario: Setting form body from .txt fixture file
        Given I mock http call to forward request body for path /users/txt
        And set request form body from text_00
        When I POST https://fake.io/users/txt
        Then response status code should be 200

    @js
    Scenario: Setting json body from .js fixture file
        Given I mock http call to forward request body for path /users/js
        And set request json body from module_00
        When I POST https://fake.io/users/js
        Then response status code should be 200

    @js
    Scenario: Setting form body from .js fixture file
        Given I mock http call to forward request body for path /users/js
        And set request form body from module_00
        When I POST https://fake.io/users/js
        Then response status code should be 200

    @json
    Scenario: Setting json body from .json fixture file
        Given I mock http call to forward request body for path /users/json
        And set request json body from json_00
        When I POST https://fake.io/users/json
        Then response status code should be 200
        And json response should fully match
            | field       | matcher | value             |
            | data        | equal   | fixture from JSON |
            | testing     | equal   | true((boolean))   |
            | awesomeness | equal   | 100((number))     |

    Scenario: Setting json body directly
        Given I mock http call to forward request body for path /users/json
        And set request json body
            | id         | 1       |
            | first_name | Raphaël |
            | last_name  | Benitte |
            | gender     | male    |
        When I POST https://fake.io/users/json
        Then response status code should be 200
        Then response body should match snapshot

    @json
    Scenario: Setting form body from .json fixture file
        Given I mock http call to forward request body for path /users/json
        And set request form body from json_00
        When I POST https://fake.io/users/json
        Then response status code should be 200

    @json @header
    Scenario: Setting json body from .json fixture file
        Given I mock http call to forward request body for path /users
        And set request json body from json_00
        And I set request headers
            | key             | value            |
            | content-type    | application/json |
            | accept-language | fr               |
            | User-Agent      | veggies/2.0      |
        When I POST https://fake.io/users
        Then response status code should be 200
        And I pick response header location as location
        And I clear request body
        And I clear request headers
        Given I assign request headers
            | params          | value |
            | accept-language | en    |
        And I set User-Agent request header to veggies/2.1
        And I mock GET http call to forward request body for path /users/1
        And I GET {{location}}
        Then response status code should be 200

    Scenario: Setting request query
        Given I mock http call to forward request body for path /users/json?query=1
        And set request form body from json_00
        And I set request query
            | query | 1 |
        When I POST https://fake.io/users/json
        Then response status code should be 200

    Scenario: Get posts from http_api/fixtures/posts.json
        Given I mock GET http call from http_api/fixtures/posts.json to forward request json body for path /posts
        When I GET http://mysite.com/api/v1/posts
        Then response status code should be 200
        And response header Connection should equal keep-alive
        And response body should match snapshot
        And I should receive a collection of 32 items for path posts

    Scenario: Follow redirect
        Given I mock GET http call from http_api/fixtures/redirect.json to forward request json body for path /posts?redirect=true
        Given I do not follow redirect
        When I GET http://mysite.com/api/v1/posts?redirect=true
        Then response status code should be 301
        And I pick response json redirect.url as url
        And I GET {{url}}
        Then response status code should be 301

    Scenario: I can create a new post from json file
        Given I mock POST http call to forward request json body for path /posts
        And set request json body from create_post
        When I POST http://mysite.com/api/v1/posts
        Then response status code should be 200
        And response json body should match snapshot
            | field       | matcher | value        |
            | post.userId | equal   | 10((number)) |
            | post.id     | equals  | 1((number))  |
