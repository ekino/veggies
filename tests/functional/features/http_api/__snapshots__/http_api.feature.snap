

exports[`Get posts from http_api/fixtures/posts.json 1.1`] = `Object {
  "posts": Array [
    Object {
      "id": 1,
      "text": "quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto",
      "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
      "userId": 1,
    },
    Object {
      "id": 2,
      "text": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
      "title": "qui est esse",
      "userId": 1,
    },
    Object {
      "id": 3,
      "text": "et iusto sed quo iure
voluptatem occaecati omnis eligendi aut ad
voluptatem doloribus vel accusantium quis pariatur
molestiae porro eius odio et labore et velit aut",
      "title": "ea molestias quasi exercitationem repellat qui ipsa sit aut",
      "userId": 1,
    },
    Object {
      "id": 4,
      "text": "ullam et saepe reiciendis voluptatem adipisci
sit amet autem assumenda provident rerum culpa
quis hic commodi nesciunt rem tenetur doloremque ipsam iure
quis sunt voluptatem rerum illo velit",
      "title": "eum et est occaecati",
      "userId": 1,
    },
    Object {
      "id": 5,
      "text": "repudiandae veniam quaerat sunt sed
alias aut fugiat sit autem sed est
voluptatem omnis possimus esse voluptatibus quis
est aut tenetur dolor neque",
      "title": "nesciunt quas odio",
      "userId": 1,
    },
    Object {
      "id": 6,
      "text": "ut aspernatur corporis harum nihil quis provident sequi
mollitia nobis aliquid molestiae
perspiciatis et ea nemo ab reprehenderit accusantium quas
voluptate dolores velit et doloremque molestiae",
      "title": "dolorem eum magni eos aperiam quia",
      "userId": 1,
    },
    Object {
      "id": 7,
      "text": "dolore placeat quibusdam ea quo vitae
magni quis enim qui quis quo nemo aut saepe
quidem repellat excepturi ut quia
sunt ut sequi eos ea sed quas",
      "title": "magnam facilis autem",
      "userId": 1,
    },
    Object {
      "id": 8,
      "text": "dignissimos aperiam dolorem qui eum
facilis quibusdam animi sint suscipit qui sint possimus cum
quaerat magni maiores excepturi
ipsam ut commodi dolor voluptatum modi aut vitae",
      "title": "dolorem dolore est ipsam",
      "userId": 1,
    },
    Object {
      "id": 9,
      "text": "consectetur animi nesciunt iure dolore
enim quia ad
veniam autem ut quam aut nobis
et est aut quod aut provident voluptas autem voluptas",
      "title": "nesciunt iure omnis dolorem tempora et accusantium",
      "userId": 1,
    },
    Object {
      "id": 10,
      "text": "quo et expedita modi cum officia vel magni
doloribus qui repudiandae
vero nisi sit
quos veniam quod sed accusamus veritatis error",
      "title": "optio molestias id quia eum",
      "userId": 1,
    },
    Object {
      "id": 11,
      "text": "delectus reiciendis molestiae occaecati non minima eveniet qui voluptatibus
accusamus in eum beatae sit
vel qui neque voluptates ut commodi qui incidunt
ut animi commodi",
      "title": "et ea vero quia laudantium autem",
      "userId": 2,
    },
    Object {
      "id": 12,
      "text": "itaque id aut magnam
praesentium quia et ea odit et ea voluptas et
sapiente quia nihil amet occaecati quia id voluptatem
incidunt ea est distinctio odio",
      "title": "in quibusdam tempore odit est dolorem",
      "userId": 2,
    },
    Object {
      "id": 13,
      "text": "aut dicta possimus sint mollitia voluptas commodi quo doloremque
iste corrupti reiciendis voluptatem eius rerum
sit cumque quod eligendi laborum minima
perferendis recusandae assumenda consectetur porro architecto ipsum ipsam",
      "title": "dolorum ut in voluptas mollitia et saepe quo animi",
      "userId": 2,
    },
    Object {
      "id": 14,
      "text": "fuga et accusamus dolorum perferendis illo voluptas
non doloremque neque facere
ad qui dolorum molestiae beatae
sed aut voluptas totam sit illum",
      "title": "voluptatem eligendi optio",
      "userId": 2,
    },
    Object {
      "id": 15,
      "text": "reprehenderit quos placeat
velit minima officia dolores impedit repudiandae molestiae nam
voluptas recusandae quis delectus
officiis harum fugiat vitae",
      "title": "eveniet quod temporibus",
      "userId": 2,
    },
    Object {
      "id": 16,
      "text": "suscipit nam nisi quo aperiam aut
asperiores eos fugit maiores voluptatibus quia
voluptatem quis ullam qui in alias quia est
consequatur magni mollitia accusamus ea nisi voluptate dicta",
      "title": "sint suscipit perspiciatis velit dolorum rerum ipsa laboriosam odio",
      "userId": 2,
    },
    Object {
      "id": 17,
      "text": "eos voluptas et aut odit natus earum
aspernatur fuga molestiae ullam
deserunt ratione qui eos
qui nihil ratione nemo velit ut aut id quo",
      "title": "fugit voluptas sed molestias voluptatem provident",
      "userId": 2,
    },
    Object {
      "id": 18,
      "text": "eveniet quo quis
laborum totam consequatur non dolor
ut et est repudiandae
est voluptatem vel debitis et magnam",
      "title": "voluptate et itaque vero tempora molestiae",
      "userId": 2,
    },
    Object {
      "id": 19,
      "text": "illum quis cupiditate provident sit magnam
ea sed aut omnis
veniam maiores ullam consequatur atque
adipisci quo iste expedita sit quos voluptas",
      "title": "adipisci placeat illum aut reiciendis qui",
      "userId": 2,
    },
    Object {
      "id": 20,
      "text": "qui consequuntur ducimus possimus quisquam amet similique
suscipit porro ipsam amet
eos veritatis officiis exercitationem vel fugit aut necessitatibus totam
omnis rerum consequatur expedita quidem cumque explicabo",
      "title": "doloribus ad provident suscipit at",
      "userId": 2,
    },
    Object {
      "id": 21,
      "text": "repellat aliquid praesentium dolorem quo
sed totam minus non itaque
nihil labore molestiae sunt dolor eveniet hic recusandae veniam
tempora et tenetur expedita sunt",
      "title": "asperiores ea ipsam voluptatibus modi minima quia sint",
      "userId": 3,
    },
    Object {
      "id": 22,
      "text": "eos qui et ipsum ipsam suscipit aut
sed omnis non odio
expedita earum mollitia molestiae aut atque rem suscipit
nam impedit esse",
      "title": "dolor sint quo a velit explicabo quia nam",
      "userId": 3,
    },
    Object {
      "id": 23,
      "text": "veritatis unde neque eligendi
quae quod architecto quo neque vitae
est illo sit tempora doloremque fugit quod
et et vel beatae sequi ullam sed tenetur perspiciatis",
      "title": "maxime id vitae nihil numquam",
      "userId": 3,
    },
    Object {
      "id": 24,
      "text": "enim et ex nulla
omnis voluptas quia qui
voluptatem consequatur numquam aliquam sunt
totam recusandae id dignissimos aut sed asperiores deserunt",
      "title": "autem hic labore sunt dolores incidunt",
      "userId": 3,
    },
    Object {
      "id": 25,
      "text": "ullam consequatur ut
omnis quis sit vel consequuntur
ipsa eligendi ipsum molestiae et omnis error nostrum
molestiae illo tempore quia et distinctio",
      "title": "rem alias distinctio quo quis",
      "userId": 3,
    },
    Object {
      "id": 26,
      "text": "similique esse doloribus nihil accusamus
omnis dolorem fuga consequuntur reprehenderit fugit recusandae temporibus
perspiciatis cum ut laudantium
omnis aut molestiae vel vero",
      "title": "est et quae odit qui non",
      "userId": 3,
    },
    Object {
      "id": 27,
      "text": "eum sed dolores ipsam sint possimus debitis occaecati
debitis qui qui et
ut placeat enim earum aut odit facilis
consequatur suscipit necessitatibus rerum sed inventore temporibus consequatur",
      "title": "quasi id et eos tenetur aut quo autem",
      "userId": 3,
    },
    Object {
      "id": 28,
      "text": "non et quaerat ex quae ad maiores
maiores recusandae totam aut blanditiis mollitia quas illo
ut voluptatibus voluptatem
similique nostrum eum",
      "title": "delectus ullam et corporis nulla voluptas sequi",
      "userId": 3,
    },
    Object {
      "id": 29,
      "text": "odit magnam ut saepe sed non qui
tempora atque nihil
accusamus illum doloribus illo dolor
eligendi repudiandae odit magni similique sed cum maiores",
      "title": "iusto eius quod necessitatibus culpa ea",
      "userId": 3,
    },
    Object {
      "id": 30,
      "text": "alias dolor cumque
impedit blanditiis non eveniet odio maxime
blanditiis amet eius quis tempora quia autem rem
a provident perspiciatis quia",
      "title": "a quo magni similique perferendis",
      "userId": 3,
    },
    Object {
      "id": 31,
      "text": "debitis eius sed quibusdam non quis consectetur vitae
impedit ut qui consequatur sed aut in
quidem sit nostrum et maiores adipisci atque
quaerat voluptatem adipisci repudiandae",
      "title": "ullam ut quidem id aut vel consequuntur",
      "userId": 4,
    },
    Object {
      "id": 32,
      "text": "deserunt eos nobis asperiores et hic
est debitis repellat molestiae optio
nihil ratione ut eos beatae quibusdam distinctio maiores
earum voluptates et aut adipisci ea maiores voluptas maxime",
      "title": "doloremque illum aliquid sunt",
      "userId": 4,
    },
  ],
}`;

exports[`I can create a new post from json file 1.1`] = `Object {
  "post": Object {
    "id": "equals(1((number)))",
    "text": "quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto",
    "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
    "userId": "equal(10((number)))",
  },
}`;

exports[`Setting json body directly 1.1`] = `Object {
  "first_name": "RaphaÃ«l",
  "gender": "male",
  "id": "1",
  "last_name": "Benitte",
}`;

exports[`Setting json body from .yaml fixture file 1.1`] = `Object {
  "first_name": "RaphaÃ«l",
  "gender": "male",
  "id": 1,
  "last_name": "Benitte",
}`;
